### ISSUE TITLE
No script injection when the url contains a querystring to a config file 

### ISSUE THREAD

**Issue by declanmagee** [BrowserSync/browser-sync#1092]
### Issue details

I need to pass json data to a page via a query string like below.

`http://localhost:3002/index.html?someConfig=SOMECONFIG.json` (_doesn't work_)
`http://localhost:3002/index.html?someConfig=SOMECONFIG` (_works_)

Unfortunately , this breaks browser sync due to the file extension in the query string. Can this be fixed?

### Steps to reproduce/test case

Test browserSync on any page and add a querystring to a config file with a file extension.
e.g.http://localhost:3002/index.html?someConfig=SOMECONFIG.json

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [ ^2.12.5  ]
- Node        [ 4.4.3   ]
- Npm         [  2.15.1  ]

### Affected platforms
- [x] windows

### Browsersync use-case
- [x] Gulp

### for all other use-cases, (gulp, grunt etc), please show us exactly how you're using Browsersync

```js
var gulp        = require('gulp');
var browserSync = require('browser-sync').create();
var sass        = require('gulp-sass');

var src = {
    scss: 'widgets/shared/polymer/templates/**/*.scss',
    css:  'widgets/shared/polymer/templates',
    html: 'build/**/*.html'
};

// Static Server + watching scss/html files
gulp.task('serve', ['sass'], function() {

    browserSync.init({
        server: "."
    });

    gulp.watch(src.scss, ['sass']);
    gulp.watch(src.html).on('change', browserSync.reload);
});

// Compile sass into CSS
gulp.task('sass', function() {
    return gulp.src(src.scss)
        .pipe(sass())
        .pipe(gulp.dest(src.css))
        .pipe(browserSync.reload({stream: true}));
});

gulp.task('default', ['serve']);
```

**Comment by shakyShane** [comment]
as it's in the query, it shouldn't affect the script injection, sounds like a bug to me. I will look into it

**Comment by shakyShane** [comment]
thanks for reporting :)

**Comment by shakyShane** [comment]
Published at browser-sync@2.12.6

**Comment by declanmagee** [comment]
Great stuff,

Thanks for the quick turn around - works perfectly now :-)

Declan

On Tue, May 10, 2016 at 6:57 AM, Shane Osbourne <notifications@github.com>
wrote:

> Published at browser-sync@2.12.6
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/BrowserSync/browser-sync/issues/1092#issuecomment-218066043>
>



-- 
*Declan Magee*
*Front End Web Developer*
Dublin | Warrenpoint
Ireland
@declanmagee <http://twitter.com/declanmagee>

**Comment by pleek91** [comment]
Curios, would this also effect files with a hash (#) at the end? For example an svg symbol like like `myfile.svg#mysvg`.
